### PR TITLE
CrossFade disabling feature

### PR DESCRIPTION
**Pull request by AlexBlokh**
#1011 issue

### CODE REVIEW

File: drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java [hunk 1]
```diff
@@ -373,7 +375,7 @@ public GenericDraweeHierarchyBuilder setOverlays(List<Drawable> overlays) {
    * @return modified instance of this builder
    */
   public GenericDraweeHierarchyBuilder setOverlay(Drawable overlay) {
-    mOverlays = Arrays.asList(overlay);
+    mOverlays = Collections.singletonList(overlay);
```
Comment (AlexBlokh) [inline drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java]: Suggested by the IDE

File: drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java [hunk 2]
```diff
@@ -347,7 +349,7 @@ public GenericDraweeHierarchyBuilder setBackgrounds(List<Drawable> backgrounds)
    * @return modified instance of this builder
    */
   public GenericDraweeHierarchyBuilder setBackground(Drawable background) {
-    mBackgrounds = Arrays.asList(background);
+    mBackgrounds = Collections.singletonList(background);
```
Comment (AlexBlokh) [inline drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java]: Suggested by the IDE

Comment (plamenko) [inline drawee/src/main/java/com/facebook/drawee/generic/GenericDraweeHierarchyBuilder.java]: Unrelated change. Please make a separate pull request.

**Comment by plamenko** [comment]
There are a couple of things going on here.
* FadeDrawable already supports independently fading each layer. Therefore, no changes are necessary to it.
* As for the GenericDraweeHierarchy changes, it is not clear what is the fading semantics with crossfade disabled.

Can you give us more context on what are you trying to achieve actually? If you do not wish placeholder to be faded out when fading to the actual image, you can set it as a background instead. That way it will stay faded in. We will be able to help once you provide more context on what behavior would you liek to achieve.

**Comment by AlexBlokh** [comment]
As I mentioned in #1011 I want to achieve the same behaviour as TransitionDrawalbe.crossfadeEnabled(false).

When new image will be crossfaded in, the foreground one would become invisible, reducing overdrawing

I'll review the feedback and update the pull request

**Comment by plamenko** [comment]
Thanks! I believe you can actually achieve what you want without any changes to Fresco. See my comment on the issue.
